### PR TITLE
F/abstract firebase lib

### PR DESCRIPTION
**Pull request by philals**
How this for abstraction? Be ruthless on my javascriping. Or let me know if you would have done it different.

I could not figure out how to put the mixins into the fireBaseWrapper class rather than the game-table.js

### CODE REVIEW

**Comment by philals** [comment]
This is just abstracting out the Firebase stuff into it's own class as we were dealing with 2 URLs now. I did not read the docs. I think that means we needed 2x "new Firebase(fbPath);"

**Comment by pjho** [comment]
Hey man, sorry for delay on replying to this. Have gone through it a few times to see if I'd do it differently. I like the abstraction and the single interface to the api in our game-table. Main comments:
- I think componentWillMount is the most 'correct' (in terms of react component lifecycle) place to instantiate the Firebase helper and it would be best to attach it as a property on the react class - so this.firebase when instantiating and referencing
- It'd be good to add update & delete player methods but I should refactor my code for this - update & delete player should be methods on the player-table class and passed down to the child components through props.. so I'll have a look at doing this. 

The mixin is a specific addon to the react class - its a class which contains react methods and additional  methods which are merged with the react class - so I think it would be hard to abstract out. I think you could make our abstraction as a mixin, extend the reactfire mixin then add it to our component as a mixin. However react is moving away from using mixins so don't think it's worth the time to try and work out how to do that :)

Anyhow, other than that, looks good. I can't merge as is, I think because I pushed some changes before looking at this so you might need to pull those in and push up again.

File: src/utils/FirebaseLib.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import ReactFire from 'reactfire';
```
Comment (pjho) [inline src/utils/FirebaseLib.js]: `import ReactFire from 'reactfire';`

This should be `import Firebase from 'firebase';` no? ReactFire is the mixin..

**Comment by philals** [comment]
Closing as new Git flow

**Pull request closed without merging**